cpu/efm32/periph/uart.c

```diff
@@ -154,6 +154,42 @@ int uart_init(uart_t dev, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     return 0;
 }
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+int uart_mode(uart_t dev, uart_data_bits_t data_bits, uart_parity_t parity,
+              uart_stop_bits_t stop_bits)
+{
+    if (parity == UART_PARITY_MARK || parity == UART_PARITY_SPACE) {
+        return UART_NOMODE;
+    }
+
+#ifdef USE_LEUART
+    if (_is_usart(dev)) {
+#endif
+        USART_TypeDef *uart = (USART_TypeDef *) uart_config[dev].dev;
+
+        USART_FrameSet(uart,
+                       USART_DataBits2Def(data_bits),
+                       USART_StopBits2Def(stop_bits),
+                       USART_Parity2Def(parity));
+#ifdef USE_LEUART
+    } else {
+        if (data_bits != UART_DATA_BITS_8) {
+            return UART_NOMODE;
```

Comment:
Why though?
`USART_DataBits2Def()` has `EFM_ASSERT(bits >= 4 && bits <= 16);`.

Reply:
The EMLIB api allows up to 16 data bits: http://devtools.silabs.com/dl/documentation/doxygen/5.7/efr32mg12/html/group__USART.html#ga882a4def49cdb2fb18622d61b24eeacd. However, they are not available for U(S)ART operation.

Reply:
Sorry, I misread. LEUART only supports 8 or 9 data bits, and since 9 data bits is not supported by RIOT-OS, the only option is 8 data bits. This is checked by [LEUART_DataBits2Def](https://github.com/basilfx/RIOT-gecko-sdk/blob/test/usart_leuart_modes/dist/emlib-extra/src/em_leuart_utils.c#L16).